src/Orleans.Runtime/Catalog/Catalog.cs

```diff
@@ -746,18 +744,24 @@ public bool TryGetActivationData(ActivationId activationId, out ActivationData d
             return data != null;
         }
 
-        private Task DeactivateActivationsFromCollector(List<ActivationData> list)
+        private async Task DeactivateActivationsFromCollector(List<ActivationData> list)
         {
+            var cts = new CancellationTokenSource(this.collectionOptions.Value.DeactivationTimeout);
+            var taskCompletionSourceList = new TaskCompletionSource<object>[list.Count];
```

Comment:
Does it make sense to use the `MultiTaskCompletionSource` here instead? It ought to be cheaper/faster, but I don't know if there are semantics from having individual TCS' which would help